zapier/src/creates/measurement.ts

```diff
@@ -0,0 +1,119 @@
+import { baseURL } from "../utils";
+import {
+  type HttpRequestOptions,
+  type Bundle,
+  type ZObject,
+} from "zapier-platform-core";
+
+export type MeasurementInputData = {
+  hardwareId: string;
+  testId: string;
+  name: string;
+  type: string;
+  data: string;
+  pass: boolean;
+};
+
+const inputFieldUrl = `${baseURL}/api/zapier/measurement-fields`;
+const inputFields = async (z: ZObject, bundle: Bundle) => {
+  const options: HttpRequestOptions = {
+    method: "GET",
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${bundle.authData.access_token}`,
+    },
+    params: {},
+  };
+
+  return z.request(inputFieldUrl, options).then((response) => {
+    response.throwForStatus();
+    const results = response.json as Array<Record<string, string>>;
+    return results;
+  });
+};
+
+const performURL = `${baseURL}/api/v1/measurements`;
+const perform = async (z: ZObject, bundle: Bundle<MeasurementInputData>) => {
+  const options: HttpRequestOptions = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      Authorization: `Bearer ${bundle.authData.access_token}`,
+    },
+    params: {},
+    body: {
+      hardwareId: bundle.inputData.hardwareId,
+      testId: bundle.inputData.testId,
+      name: bundle.inputData.name,
+      data: {
+        type: bundle.inputData.type,
+        value:
+          bundle.inputData.type === "boolean"
+            ? Boolean(bundle.inputData.data)
+            : (JSON.parse(bundle.inputData.data) as Record<
+                string,
+                number[] | string[]
+              >),
```

Comment:
this is not very easy to maintain when adding more measurement types (i already did a few days ago when adding scalar), should extract this into a function